test/dynamo/test_higher_order_ops.py

```diff
@@ -3289,11 +3336,7 @@ def fn(x, y):
 
         actual = fn(x, y)
         expected = torch.compile(fn, backend="aot_eager", fullgraph=False)(x, y)
-        self.assertEqual(len(counters["graph_break"]), 1)
-        self.assertEqual(
-            dict(counters["graph_break"]),
-            {"NYI - torch.func.vmap: kwargs arguments are currently unsupported.": 2},
-        )
+        self.assertEqual(len(counters["graph_break"]), 0)
```

Comment:
@guilhermeleobas: We should add some explicit tests that check torch.compile(backend="aot_eager") where the function being vmapped over has graph breaks in it. I am not sure this situation works (in theory, it probably doesn't).

Here's an example:
```py
def g(x):
   y = x.sum(0)
   print("a")  # cause a graph break
   z = y.sum(0)

@torch.compile(backend="aot_eager")
def f(x):
    return vmap(g)(x)

x = torch.randn(2, 3, 4, 5)
f(x)
```
I would assume this does not work because AOTAutograd doesn't know what to do when Dynamo passes it a graph that has a BatchedTensor input or output -- it only knows what to do in the situation where Dynamo passes it a graph whose input and outputs are regular Tensors.

In this situation, for now, we would want the entire of vmap(g)(x) to be run in eager-mode PyTorch. Open question on how to actually get there.

Reply:
Bounced some ideas with Voz. The TL;DR:
1. If there is a graph break, what we want to do is apply torch._dynamo.disable to vmap(g), and then carry on with the graph break. This will prevent Dynamo from looking into that specific vmap(g) again (vmap(g) returns a new function every time it is called)
2. A user might do something like the following, where they save the result of vmap(f):
```
vf = vmap(f)
vf(x)
vf(y)
```
However, this ends up with the problem that if vf(x) graph breaks and falls back to eager-mode PyTorch, we do not necessarily want vf(y) to graph break. One way to resolve this is to have vf(x) actually generate fresh new function `k` every time it is called and invoke k(x), and then the thing we actually torch._dynamo.disable is `k`.

TBH we can do (2) as a follow-up, but this PR should have (1) at the least

Reply:
I think I got it! There's a `torch._C._dynamo.eval_frame.skip_code` function that one can mark a frame to be skipped. This avoids Dynamo from tracing the frame again in case of a graph break. Will add a couple of tests in a bit.